.github/workflows/kind_integration.yml

```diff
@@ -125,35 +125,11 @@ jobs:
 
         # Validate the CLI version matches the current build tag.
         [[ "$TAG" == "$($HOME/.linkerd version --short --client)" ]]
-    - name: Setup default KinD cluster
-      if: matrix.integration_test != 'custom_domain'
-      run: bin/kind create cluster --wait 300s
-    - name: Setup custom_domain KinD cluster
-      if: matrix.integration_test == 'custom_domain'
-      run: bin/kind create cluster --wait 300s --config test/testdata/custom_cluster_domain_config.yaml
-    - name: Load image archives into the local KinD cluster
+    - name: Run integration tests
       if: github.event_name == 'push' || !github.event.pull_request.head.repo.fork
-      env:
-        PROXY_INIT_IMAGE_NAME: gcr.io/linkerd-io/proxy-init:v1.3.1
-        PROMETHEUS_IMAGE_NAME: prom/prometheus:v2.15.2
       run: |
-        # Fetch images from the Packet host and load them into the local KinD cluster
-        bin/kind-load --images --images-host ssh://linkerd-docker
-
-        # Load proxy-init and prometheus images into KinD while it is
-        # available. Allow these commands to fail since they will be cached
-        # for the next run.
-        kind load image-archive <(DOCKER_HOST=ssh://linkerd-docker docker save $PROXY_INIT_IMAGE_NAME) 2>&1 || true
-        kind load image-archive <(DOCKER_HOST=ssh://linkerd-docker docker save $PROMETHEUS_IMAGE_NAME) 2>&1 || true
-    - name: Load image archives into the local KinD cluster (Forked repositories)
+        bin/tests --images --images-host ssh://linkerd-docker --name ${{ matrix.integration_test }} $HOME/.linkerd
```

Comment:
If $HOME would contain whitespaces, there would be a failure here.

`"$HOME/.linkerd"` or `"$HOME"/.linkerd` would be the safest choice eliminating unlikely corner case issues, IMO.